src/main/flight/pid.c

```diff
@@ -130,6 +130,91 @@ float getdT (void) {
+
+    // get raw stick positions (-500 to 500):
+    const int32_t stickPosAil = getRcStickDeflection(FD_ROLL, rxConfigMidrc);
+    const int32_t stickPosEle = getRcStickDeflection(FD_PITCH, rxConfigMidrc);
+
+    // 0 at center stick, 500 at max stick deflection:
+    const int32_t mostDeflectedPos = MAX(ABS(stickPosAil), ABS(stickPosEle));
+
+    // start with 100 at center stick, 0 at max stick deflection:
+    horizonLevelStrength = (500 - mostDeflectedPos) / 5;
+
+    // 0 at level, 900 at vertical, 1800 at inverted (degrees * 10)
+    currentInclination = MAX(ABS(attitude.values.roll),
+                                                ABS(attitude.values.pitch));
+
+    // 'horizon_incl_fact' configuration setting:
+    // 0-99 = range 1 (leveling always active when sticks centered)
+    // 100-250 = range 2 (leveling can be totally off when inverted)
+    if (horizonInclFact >= 100) {
+        // range 2 (leveling can be totally off when inverted)
+        if (horizonInclFact < 250) {
+            // horizon_incl_fact 100 to 200 => 2700 to 900
+            //  (represents where leveling goes to zero):
+            cutoffDeciDegrees = (250-horizonInclFact) * 18;
+            // inclinationLevelRatio (0 to 100) is smaller (less leveling)
+            //  for larger inclinations; 0 at cutoffDeciDegrees value:
+            inclinationLevelRatio = constrain(
+                              (((cutoffDeciDegrees-currentInclination)*10) /
+                                           (cutoffDeciDegrees/10)), 0, 100);
+            // apply configured horizon sensitivity:
+            if (horizonSensit <= 0) {       // zero means no leveling
+                horizonLevelStrength = 0;
+            } else {
+                // when stick is near center (horizonLevelStrength ~= 100)
+                //  H_sensitivity value has little effect,
+                // when stick is deflected (horizonLevelStrength near 0)
+                //  H_sensitivity value has more effect:
+                horizonLevelStrength = constrain(
+                                        (((horizonLevelStrength-100) * 100 /
+                                             horizonSensit) + 100), 0, 100);
+            }
+            // apply inclination ratio, which may lower leveling
+            //  to zero regardless of stick position:
+            horizonLevelStrength =
+                         horizonLevelStrength * inclinationLevelRatio / 100;
+        }
+        else
+          horizonLevelStrength = 0;
+    } else {       // horizon_incl_fact < 100
+        // range 1 (leveling always active when sticks centered)
+        if (horizonInclFact > 0) {
+            // ratio of 100 to 0 (larger means more leveling):
+            factorRatio = 100 - horizonInclFact;
+            // inclinationLevelRatio (0 to 100) is smaller (less leveling)
+            //  for larger inclinations, goes to 100 at inclination level:
+            inclinationLevelRatio =
```

Comment:
@martinbudden  That rework of line 194 is not equivalent.  For instance, if horizonTiltEffect=75, factorRatio=25 and currentInclination=300:

old:  ((1800-300)/18 * (100-25)) / 100 + 25 = 87

new:  100 + 75 * ((1800-300)/1800)  - 1 = 99

I'm not seeing a good way to simplify the line.  I will, however, add "const int" to line 191.

--ET